src/Control/HTTPRequest.php

```diff
@@ -434,8 +434,7 @@ public function isAjax()
      * @param string $offset
      * @return bool
      */
-    #[\ReturnTypeWillChange]
-    public function offsetExists($offset)
+    public function offsetExists(mixed $offset): bool
```

Comment:
You've marked `$offset` as mixed but the phpdoc says `string` - which is correct?
It may be best to just leave the argument loosely typed instead of strongly typing it as `mixed` in the case that it _should_ be `string` but sometimes isn't. That will make it easier on us when/if we do eventually go through and strongly type everything.

My guess is that we should just use the typing that PHP suggests (i.e. mixed as you have done) in which case we should remove the incorrect PHPdoc.

Reply:
This applies to other arguments in this and related PRs as well.

Reply:
The parent class won't allow me to explicitly say `string`, but we know that the value in this context should always be a string.

In the past, we've used the PHPDoc to provide more specific definitions or definition that work better for type hinting. e.g.: If I know that a method will return an DataList of SiteTree, we would often say in the PHPDoc that it would return `DataList|SitreTree[]`.

If you think having the PHPDoc say its a string is too much of a contradiction, I'm happy to only use the explicit return type.

Reply:
If the type _should_ actually be string (i.e. only strings are valid) then we should keep the phpdoc and _not_ include the `mixed` type in the parameter. PHP won't complain about this.

If the type _can legitimately be anything_ then we should include the `mixed` type in the parameter and remove the phpdoc.

Reply:
`public function offsetExists(string $offset): bool` will throw an PHP Fatal Error, so there's no way around that.

I guess the intention of interface is that if `offsetExists` get passed some nonsense, it should behave as if the offset doesn't exists. Which our implementation will be doing.

At the same time if you give it anything other than a string, you probably have a logic error in your code.

I think this is an extremely minor concern. I'll drop the `@param` annotation.

Reply:
> `public function offsetExists(string $offset): bool` will throw an PHP Fatal Error, so there's no way around that.

That's why that wasn't one of the two options I gave. :stuck_out_tongue:

Reply:
Misunderstood your suggestion.

I'm not sure there's any difference between `public function offsetExists(mixed $offset): bool` and `public function offsetExists($offset): bool`. My guess is PHP implicitly assigned the `mixed` type in the later case.

Reply:
Fixed.

Reply:
There's no technical difference - but there is a difference in how it's read by humans if there's a corresponding PHPDoc.

I read this as meaning "the correct type is string"
```php
/**
 * @param string $someVar
 */
function somefunc($someVar) {
```

But I read this as meaning "Any type is fine, the phpdoc is outdated"
```php
/**
 * @param string $someVar
 */
function somefunc(mixed $someVar) {
```